### PR TITLE
[RD] Donor Section Tests

### PR DESCRIPTION
**Pull request by bdvorachek**
WI: https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07B0000008F7oRIAS/view

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# Features Intended for Future Release

# Features for Elevate Customers

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: tests/__mocks__/apex/data/recurringDataAccountResponse.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "DonorType": "Account",
```
Comment (lparrott) [inline tests/__mocks__/apex/data/recurringDataAccountResponse.json:2]: I see that we include these mock responses twice in different directories, is that required for these tests?

File: src/lwc/rd2EntryFormDonorSection/rd2EntryFormDonorSection.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ export default class rd2EntryFormDonorSection extends LightningElement {
     // These are exposed to the parent component
     @api parentId;
     @api recordId;
-    @api parentSObjectType
+    @api parentSObjectType;
```
Comment (lparrott) [inline src/lwc/rd2EntryFormDonorSection/rd2EntryFormDonorSection.js:30]: 😲

File: src/lwc/rd2EntryFormDonorSection/__tests__/rd2EntryFormDonorSection.test.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+        expect(lastEventDetail).toBe(FAKE_ACCOUNT_ID);
+    });
+
+    it('populates account lookup', () => {
+        const accountLookup = getAccountLookup(element);
+        expect(accountLookup).toBeTruthy();
+        expect(accountLookup.value).toBe(FAKE_ACCOUNT_ID);
+    });
+
+    it('clears account lookup on donor type change', async () => {
```
Comment (lparrott) [inline src/lwc/rd2EntryFormDonorSection/__tests__/rd2EntryFormDonorSection.test.js:159]: I *think* this is testing the following AC, but I want to confirm: "When a user selects Account for Donor Type And they enter an Account And they switch back to Contact for Donor Type Then the Account value is cleared"

**Review by lparrott — COMMENTED** [review]
@bdvorachek Looks great, I just had a few questions/comments.

File: tests/__mocks__/apex/data/recurringDataAccountResponse.json (hunk 1 above)
Comment (bdvorachek) [reply tests/__mocks__/apex/data/recurringDataAccountResponse.json:2]: I think so. The only alternative I was able to come up with would be moving it up the directory structure, something like `require(../../../data/recurringDataAccountResponse);`

File: src/lwc/rd2EntryFormDonorSection/__tests__/rd2EntryFormDonorSection.test.js (hunk 3 above)
Comment (bdvorachek) [reply src/lwc/rd2EntryFormDonorSection/__tests__/rd2EntryFormDonorSection.test.js:159]: I don't think I have that particular case here, stopped one donor type change short of it! Will add.

Comment (bdvorachek) [reply src/lwc/rd2EntryFormDonorSection/__tests__/rd2EntryFormDonorSection.test.js:166]: Added!

**Review by lparrott — APPROVED** [review]

**Review by dfuller114 — APPROVED** [review]
looks good! thanks Ben!

**Pull request merged**